### PR TITLE
UnitTest_1428

### PR DESCRIPTION
**Pull request by forestik**

### CODE REVIEW

File: core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    private FactOfTheDayTranslationServiceImpl factOfTheDayTranslationService;
+    @Mock
+    private FactOfTheDayTranslationRepo factOfTheDayTranslationRepo;
+    @Mock
+    private ModelMapper modelMapper;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java:39]: Please see 
- changes in https://github.com/ita-social-projects/GreenCity/pull/1353/files
- conversation in https://github.com/ita-social-projects/GreenCity/pull/1353/files/fd146c507103b8713a72173d64e2c9df32da32fb#diff-b727cc1b5710fa06c4f437b8b7413278 (click on Show outdated)

You will need:
- make fields final, remove `Autowired` annotation
- replace constructor by `AllArgsConstructor` annotation
- use `InjectMocks` annotation and remove `init` method

File: core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        factOfTheDayTranslationService= new FactOfTheDayTranslationServiceImpl(factOfTheDayTranslationRepo, modelMapper);
+    }
+
+    @Test
+    void getFactOfTheDayById() {
+        Long id = 1L;
+        FactOfTheDayTranslation factOfTheDayTranslation = new FactOfTheDayTranslation();
+        factOfTheDayTranslation.builder()
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java:47]: What is this test checking if the result of the `build()` invocation is ignored?

File: core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                .factOfTheDay(ModelUtils.getFactOfTheDay())
+                .build();
+        when(factOfTheDayTranslationRepo.save(factOfTheDayTranslation)).thenReturn(factOfTheDayTranslation);
+        assertEquals(factOfTheDayTranslation, factOfTheDayTranslationService.save(factOfTheDayTranslation));
+    }
+
+    @Test
+    void saveAll() {
+        FactOfTheDayTranslation factOfTheDayTranslation = new FactOfTheDayTranslation();
+        factOfTheDayTranslation.builder()
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java:74]: The same here.

File: core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        when(factOfTheDayTranslationRepo.saveAll(factOfTheDayTranslationList)).thenReturn(factOfTheDayTranslationList);
+        assertEquals(factOfTheDayTranslationList, factOfTheDayTranslationService.saveAll(factOfTheDayTranslationList));
+    }
+
+    @Test
+    void deleteAll() {
+        FactOfTheDayTranslation factOfTheDayTranslation = new FactOfTheDayTranslation();
+        factOfTheDayTranslation.builder()
+                .id(1L)
+                .content("Content")
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/FactOfTheDayTranslationServiceImplTest.java:90]: The same here.

File: core/src/test/java/greencity/service/impl/HabitStatusCalendarServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+class HabitStatusCalendarServiceImplTest {
+
+    HabitStatusCalendarService habitStatusCalendarService;
+    @Mock
+    private HabitStatusCalendarRepo habitStatusCalendarRepo;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
+        habitStatusCalendarService = new HabitStatusCalendarServiceImpl(habitStatusCalendarRepo);
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/HabitStatusCalendarServiceImplTest.java:38]: The same here.

File: core/src/test/java/greencity/service/impl/HabitStatusServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+    private HabitStatusRepo habitStatusRepo;
+    @Mock
+    private HabitStatusCalendarService habitStatusCalendarService;
+    @Mock
+    private ModelMapper modelMapper;
+
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/HabitStatusServiceImplTest.java:48]: The same here.

File: core/src/test/java/greencity/service/impl/SocialNetworkImageServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    SocialNetworkImageServiceImpl socialNetworkImageService;
+    @Mock
+    SocialNetworkImageRepo socialNetworkImageRepo;
+    @Mock
+    FileService fileService;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
+        socialNetworkImageService= new SocialNetworkImageServiceImpl(socialNetworkImageRepo, fileService);
```
Comment (antonbakalets) [inline core/src/test/java/greencity/service/impl/SocialNetworkImageServiceImplTest.java:30]: The same here.

**Review by antonbakalets — CHANGES_REQUESTED** [review]

**Review by antonbakalets — APPROVED** [review]

**Review by eyouu — APPROVED** [review]

**Review by Yura09 — APPROVED** [review]

**Pull request merged**